File: ts-sdk/src/sdk/models/operations/getmetadatatemplates.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetMetadataTemplatesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=metadata_instance_id" })
  metadataInstanceId: string;
}


export class GetMetadataTemplatesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetMetadataTemplatesQueryParams;
}


export class GetMetadataTemplatesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  metadataTemplates?: shared.MetadataTemplates;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/getmetadatatemplates.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetMetadataTemplatesQueryParams struct {
	MetadataInstanceID string `queryParam:"style=form,explode=true,name=metadata_instance_id"`
}

type GetMetadataTemplatesRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetMetadataTemplatesQueryParams
}

type GetMetadataTemplatesResponse struct {
	ClientError       *shared.ClientError
	ContentType       string
	MetadataTemplates *shared.MetadataTemplates
	StatusCode        int64
}
